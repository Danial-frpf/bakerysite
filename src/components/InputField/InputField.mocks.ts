import { IInputField } from ".";

const small: IInputField = {
    label: "User:",
    name: "user",
    placeholder: "placeholder...",
    size: "sm",
    type: "text",
};

const medium: IInputField = {
    label: "User:",
    name: "user",
    placeholder: "placeholder...",
    size: "md",
    type: "text",
};

const large: IInputField = {
    label: "User:",
    name: "user",
    placeholder: "placeholder...",
    size: "lg",
    type: "text",
};

const extraLarge: IInputField = {
    label: "User:",
    name: "user",
    placeholder: "placeholder...",
    size: "xl",
    type: "text",
};

export const mockInputFieldProps = {
    small,
    medium,
    large,
    extraLarge,
};
