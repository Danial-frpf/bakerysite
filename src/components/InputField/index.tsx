import { useState } from "react";
import Label from "../Label";
import styles from "./styles.module.css";

export interface IInputField {
    label: string;
    name: string;
    placeholder?: string;
    type?: "text" | "password" | "file";
    size?: "sm" | "md" | "lg" | "xl";
    props?: React.DetailedHTMLProps<
        React.HTMLAttributes<HTMLDivElement>,
        HTMLDivElement
    >;
}
const InputField: React.FC<IInputField> = ({
    label,
    name,
    type = "text",
    size = "md",
    placeholder = "",
    props,
}) => {
    const [value, setValue] = useState("");
    const changeHandler = (e: React.FormEvent<HTMLInputElement>) => {
        setValue(e.currentTarget.value);
    };

    let sizeStyle = "";

    switch (size) {
        case "sm":
            sizeStyle = styles.sm;
            break;
        case "md":
            sizeStyle = styles.md;
            break;
        case "lg":
            sizeStyle = styles.lg;
            break;
        case "xl":
            sizeStyle = styles.xl;
            break;
        default:
            break;
    }

    return (
        <div
            data-testid="InputField"
            {...props}
            className={`${styles.container} ${sizeStyle} ${props?.className}`}
        >
            <Label label={label} size={size} />
            <input
                type={type}
                name={name}
                placeholder={placeholder}
                value={value}
                onChange={changeHandler}
            />
        </div>
    );
};

export default InputField;
