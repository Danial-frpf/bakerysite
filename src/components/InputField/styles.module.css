.container {
    display: grid;
    grid-template-columns: 1fr 2fr;
    text-align: center;
}

.container input {
    min-width: 64px;
    padding: 0.5vh 0.5vw;
    border: 2px solid var(--main-800);
    background-color: var(--secondary-orange-40);
}

.container input:focus {
    background-color: var(--secondary-yellow-40);
    border-color: var(--secondary-yellow);
}

.sm {
    margin-bottom: 0.5vh;
}

.md {
    margin-bottom: 1vh;
}

.lg {
    margin-bottom: 1vh;
}

.xl {
    margin-bottom: 2vh;
}

.sm input {
    font-size: var(--baseFont);
}

.md input {
    font-size: var(--h4);
}

.lg input {
    font-size: var(--h2);
}

.xl input {
    font-size: var(--h1);
}

@media (max-width: 512px) {
    .container {
        text-align: left;
        grid-template-columns: 1fr;
    }
}
