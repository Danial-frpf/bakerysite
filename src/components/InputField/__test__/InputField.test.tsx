import { createRoot } from "react-dom/client";
import { act } from "react-dom/test-utils";
import InputField from "..";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);
it("renders correctly without crashing", () => {
    const div = document.createElement("div");
    act(() => {
        createRoot(div!).render(
            <InputField label="username" name="username"></InputField>
        );
    });
});

it("renders correctly without crashing", () => {
    const div = document.createElement("div");
    act(() => {
        createRoot(div!).render(
            <InputField label="username" name="username" size="sm"></InputField>
        );
    });
});

it("renders correctly without crashing", () => {
    const div = document.createElement("div");
    act(() => {
        createRoot(div!).render(
            <InputField label="username" name="username" size="lg"></InputField>
        );
    });
});

it("renders InputField correctly", () => {
    const { getByTestId } = render(
        <InputField label="username" name="username"></InputField>
    );
    expect(getByTestId("InputField")).toHaveTextContent("username");
});

it("renders InputField correctly", () => {
    const { getByTestId } = render(
        <InputField label="username" name="username" size="sm"></InputField>
    );
    expect(getByTestId("InputField")).toHaveTextContent("username");
});

it("renders InputField correctly", () => {
    const { getByTestId } = render(
        <InputField label="username" name="username" size="lg"></InputField>
    );
    expect(getByTestId("InputField")).toHaveTextContent("username");
});

it("matches snapshot", () => {
    const tree = renderer
        .create(<InputField label="username" name="username"></InputField>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("matches snapshot", () => {
    const tree = renderer
        .create(
            <InputField label="username" name="username" size="sm"></InputField>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("matches snapshot", () => {
    const tree = renderer
        .create(
            <InputField label="username" name="username" size="lg"></InputField>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
