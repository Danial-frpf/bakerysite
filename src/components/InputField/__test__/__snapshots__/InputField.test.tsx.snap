// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container md undefined"
  data-testid="InputField"
>
  <h4
    data-testid="Label"
  >
    <span>
      username
    </span>
  </h4>
  <input
    name="username"
    onChange={[Function]}
    placeholder=""
    type="text"
    value=""
  />
</div>
`;

exports[`matches snapshot 2`] = `
<div
  className="container sm undefined"
  data-testid="InputField"
>
  <p
    data-testid="Label"
  >
    <span>
      username
    </span>
  </p>
  <input
    name="username"
    onChange={[Function]}
    placeholder=""
    type="text"
    value=""
  />
</div>
`;

exports[`matches snapshot 3`] = `
<div
  className="container lg undefined"
  data-testid="InputField"
>
  <h2
    data-testid="Label"
  >
    <span>
      username
    </span>
  </h2>
  <input
    name="username"
    onChange={[Function]}
    placeholder=""
    type="text"
    value=""
  />
</div>
`;

exports[`matches snapshot 4`] = `
<div
  className="container xl undefined"
  data-testid="InputField"
>
  <h1
    data-testid="Label"
  >
    <span>
      username
    </span>
  </h1>
  <input
    name="username"
    onChange={[Function]}
    placeholder=""
    type="text"
    value=""
  />
</div>
`;
